File: frontend/test-api.js
// 在浏览器控制台运行这个脚本来测试API

const testAPI = async () => {
  console.log('开始测试API连接...')

  try {
    // 测试评论统计API
    console.log('\n1. 测试评论统计API:')
    const statsRes = await fetch('http://localhost:5000/api/comments/recipe/1/stats')
    console.log('Status:', statsRes.status)
    const stats = await statsRes.json()
    console.log('Stats:', stats)

    // 测试评论列表API
    console.log('\n2. 测试评论列表API:')
    const commentsRes = await fetch('http://localhost:5000/api/comments/recipe/1')
    console.log('Status:', commentsRes.status)
    const comments = await commentsRes.json()
    console.log('Comments:', comments)

    console.log('\n✅ API测试成功!')
  } catch (error) {
    console.error('\n❌ API测试失败:', error)
  }
}

// 运行测试
testAPI()

File: backend/test_api.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""测试所有API接口"""

import requests

def test_api():
    base_url = 'http://localhost:5000/api'
    
    tests = [
        ('GET', '/recipes', None),
        ('GET', '/posts', None),
        ('POST', '/auth/login', {'username': 'user1', 'password': 'user123'})
    ]
    
    for method, endpoint, data in tests:
        url = f"{base_url}{endpoint}"
        try:
            if method == 'GET':
                response = requests.get(url)
            elif method == 'POST':
                response = requests.post(url, json=data)
            
            print(f"{method} {endpoint}: {response.status_code}")
            if response.status_code != 200:
                print(f"  Error: {response.text[:200]}")
            else:
                print(f"  Success")
        except Exception as e:
            print(f"{method} {endpoint}: ERROR - {e}")

if __name__ == '__main__':
    test_api()
